src/main/java/me/botsko/prism/commands/DeleteCommand.java

```diff
@@ -7,131 +7,101 @@
-						+ " wiped the live queue before it could be written to the database. "
-						+ RecordingQueue.getQueue().size() + " events lost.");
-				RecordingQueue.getQueue().clear();
-				call.getSender().sendMessage(Prism.messenger.playerSuccess("Unwritten data in queue cleared."));
-			}
-			else {
-				call.getSender().sendMessage(Prism.messenger.playerError("Event queue is empty, nothing to wipe."));
-			}
-			return;
-		}
-
-		// Process and validate all of the arguments
-		final QueryParameters parameters = PreprocessArgs.process(plugin, call.getSender(), call.getArgs(),
-				PrismProcessType.DELETE, 1, !plugin.getConfig().getBoolean("prism.queries.never-use-defaults"));
-		if (parameters == null) {
-			return;
-		}
-		parameters.setStringFromRawArgs(call.getArgs(), 1);
-
-		// determine if defaults were used
-		final ArrayList<String> defaultsUsed = parameters.getDefaultsUsed();
-		StringBuilder defaultsReminder = new StringBuilder();
-		if (!defaultsUsed.isEmpty()) {
-			defaultsReminder.append(" using defaults:");
-			for (final String d : defaultsUsed) {
-				defaultsReminder.append(" ").append(d);
-			}
-		}
-
-		if (parameters.getFoundArgs().size() > 0) {
-
-			call.getSender().sendMessage(Prism.messenger.playerSubduedHeaderMsg("Purging data..." + defaultsReminder));
-			call.getSender().sendMessage(Prism.messenger
-					.playerHeaderMsg("Starting purge cycle." + ChatColor.GRAY + " No one will ever know..."));
-
-			plugin.getServer().getScheduler().runTaskAsynchronously(plugin, () -> {
-				int purge_tick_delay = plugin.getConfig().getInt("prism.purge.batch-tick-delay");
-				if (purge_tick_delay < 1) {
-					purge_tick_delay = 20;
-				}
-
-				// build callback
-				final SenderPurgeCallback callback = new SenderPurgeCallback();
-				callback.setSender(call.getSender());
-
-				// add to an arraylist so we're consistent
-				final CopyOnWriteArrayList<QueryParameters> paramList = new CopyOnWriteArrayList<>();
-				paramList.add(parameters);
-
-				final ActionsQuery aq = new ActionsQuery(plugin);
-				final long minId = aq.getMinIDForQuery(parameters);
-				final long maxId = aq.getMaxIDForQuery(parameters);
-
-				Prism.log(
-						"Beginning prism database purge cycle. Will be performed in batches so we don't tie up the db...");
-				deleteTask = plugin.getServer().getScheduler().runTaskAsynchronously(plugin,
-						new PurgeTask(plugin, paramList, purge_tick_delay, minId, maxId, callback));
-			});
-		}
-		else {
-			call.getSender().sendMessage(Prism.messenger.playerError("You must supply at least one parameter."));
-		}
-	}
-
-	@Override
-	public List<String> handleComplete(CallInfo call) {
-		return PreprocessArgs.complete(call.getSender(), call.getArgs());
-	}
+public class DeleteCommand extends AbstractCommand {
+
+    private final Prism plugin;
+    protected BukkitTask deleteTask;
+    /**
+     * Constructor.
+     *
+     * @param plugin Prism
+     */
+    public DeleteCommand(Prism plugin) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['CTOR_DEF' should be separated from previous statement.](https://app.codacy.com/gh/AddstarMC/Prism-Bukkit/pullRequest?prid=5438808)